pretix_newsletter_manual/exporters.py

```diff
@@ -29,9 +29,12 @@ def iterate_list(self, form_data):
                     headers.append(label)
         yield headers
 
+
         for r in qs.select_related("order", "order__event"):
             tz = pytz.timezone(r.order.event.settings.timezone)
-
+            possible_positions = [op for op in r.order.all_positions.all() if not op.canceled and op.attendee_name]
+            name_parts = possible_positions[0] if possible_positions else {}
```

Comment:
I know @raphaelm suggested the it, but I would not call this variable `name_parts` as this is an orderposition. I’d call it `op_with_attendee_name` and have the else `None`. Or event shorter (not tested!):

```suggestion
            op_with_attendee_name = next(op for op in r.order.all_positions.all() if not op.canceled and op.attendee_name, None)
```

Reply:
I probabyl just messed up my suggestion, I probably meant

```
name_parts = possible_positions[0].name_parts if possible_positions else {}
```